plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-product-related-products-controller.php

```diff
@@ -0,0 +1,254 @@
+<?php
+/**
+ * REST API related-products controller
+ *
+ * Handles requests to the /products/<id>/related-products endpoints.
+ *
+ * @package WooCommerce\RestApi
+ * @since   3.0.0
+ */
+
+use Automattic\WooCommerce\Utilities\I18nUtil;
+
+defined( 'ABSPATH' ) || exit;
+
+use Automattic\Jetpack\Constants;
+
+/**
+ * REST API related-products controller class.
+ *
+ * @package WooCommerce\RestApi
+ * @extends WC_REST_Product_Related_Products_V2_Controller
+ */
+class WC_REST_Product_Related_Products_Controller extends WC_REST_Products_V2_Controller {
+
+	/**
+	 * Endpoint namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace = 'wc/v3';
+
+	/**
+	 * Endpoint namespace.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'products/(?P<id>[\d]+)/related-products';
+
+	/**
+	 * Post type.
+	 *
+	 * @var string
+	 */
+	protected $post_type = 'product';
+
+	/**
+	 * Related product ids.
+	 *
+	 * @var array
+	 */
+	protected $ids = array();
+
+	/**
+	 * Related product categories.
+	 *
+	 * @var array
+	 */
+	protected $categories = array();
+
+	/**
+	 * Related product tags.
+	 *
+	 * @var array
+	 */
+	protected $tags = array();
+
+	/**
+	 * Combine the product terms (categories, tags, ...).
+	 *
+	 * @var boolean
+	 */
+	protected $combine = false;
+
+	/**
+	 * Register the routes for products.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				'args'   => array(
+					'id' => array(
+						'description' => __( 'Unique identifier for the variable product.', 'woocommerce' ),
+						'type'        => 'integer',
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::READABLE,
```

Comment:
If we are supporting the passing in of the attributes/categories/tags and such should we turn this into a `POST` instead? So that data can be passed in as `data` versus a `query`?

Reply:
I am not opposed to leaving it as a `GET` request for now.

Reply:
I believe that using GET method is appropriate in this case as we do not intend to make any changes to the data.

Reply:
Yeah, I suppose that is fine, it just seems like it may get messy when we also support the passing in of attributes, as attributes contain a key and value.

Reply:
It's a good point. It doesn't seem pretty easy to pass it in the query. right?